### PR TITLE
Story 30

### PR DESCRIPTION
**Pull request by ericdrosado**

### CODE REVIEW

File: game.rb [hunk 1]
```diff
@@ -24,21 +23,28 @@ def player_one_pick_word
   end
 
   def player_two_guess_word word
-    counter = 0
+    guess_counter = 0
```
Comment (tommcgee) [inline game.rb:26]: :+1: Nothing like some better variable names. This should probably become `incorrect_guess_counter` though I imagine?

File: game_view.rb [hunk 2]
```diff
@@ -1,9 +1,12 @@
 class GameView
 
-  attr_reader :guessed_letters
+  attr_reader :guessed_letters, :body_array
```
Comment (tommcgee) [inline game_view.rb:3]: I think instead of referencing body_array directly it would be better to have a method clarifying that this value is changed by `remove_hangman_body_part` like `get_current_hangman`.

Comment (tommcgee) [inline game_view.rb:3]: Also try not to name things based off of how they're implemented (especially when they're being accessed directly like this) you want to be printing `current_hangman_body` not `string_array`. It makes the code a lot more readable

File: game_view.rb [hunk 3]
```diff
@@ -22,4 +25,33 @@ def clear_view
     system "clear"
   end
 
+  def display_hangman_head
+    "      " + "_____\n" + "     " + "( * * )\n" +  "      " + "(___)"
```
Comment (tommcgee) [inline game_view.rb:29]: For these body part methods it would be nice to line up the different rows on new lines so you can kind of see what the part will look like from just looking at the code

File: game.rb [hunk 4]
```diff
@@ -24,21 +23,28 @@ def player_one_pick_word
...
     end_of_game = false
+    @io_handler.print(@game_view.body_array)
     while ! end_of_game
```
Comment (tommcgee) [inline game.rb:29]: This while loop is getting pretty huge. It would be worth breaking it down into methods that represent each part.

Comment (tommcgee) [reply game.rb:29]: For example `letter = get_letter_guess`

Comment (tommcgee) [reply game.rb:29]: You can decide on the best way to do this but this method is definitely too hard to follow.

File: spec/game_view_spec.rb [hunk 5]
```diff
@@ -40,4 +40,60 @@
 
   end
```
Comment (tommcgee) [inline spec/game_view_spec.rb:41]: In this file it would be cool to have some tests making calls to remove_hangman_part and then making sure the feet (or other parts) aren't displayed

File: game.rb [hunk 6]
```diff
@@ -24,21 +23,28 @@ def player_one_pick_word
   end
 
   def player_two_guess_word word
-    counter = 0
+    guess_counter = 0
     end_of_game = false
+    @io_handler.print(@game_view.body_array)
```
Comment (tommcgee) [inline game.rb:28]: The way you have this makes it difficult to test this is happening without messing with STDOUT. it would be nice if this was it's own method so you could at least make sure it was being called.

**Review by tommcgee — CHANGES_REQUESTED** [review]

File: game.rb (hunk 4 above)
Comment (NicoleCarpenter) [reply game.rb:29]: Omg, is this all one method? 🙀 

There is duplication that can be reduced, and we are going to have to do something about multiple object calls on the same line. Please read: http://alvinalexander.com/java/java-law-of-demeter-java-examples. This works, but you are creating code that is highly coupled.

File: game_view.rb (hunk 3 above)
Comment (NicoleCarpenter) [reply game_view.rb:29]: +1

File: spec/game_view_spec.rb [hunk 7]
```diff
@@ -40,4 +40,60 @@
...
 
+  describe '#display_hangman_head' do
+
+    it 'will return a string' do
+      expect(game_view.display_hangman_head).to eq "      " + "_____\n" + "     " + "( * * )\n" +  "      " + "(___)"
```
Comment (NicoleCarpenter) [inline spec/game_view_spec.rb:46]: This is kind of weird. You should not have computations in the tests. These should be the single-string results that you would expect

**Comment by NicoleCarpenter** [comment]
As an overall note, I don't like the flow of the game. 

First, I appreciate that the screen clears after you pick a word, but you should also include a clear when you first start the game, and then maybe after each turn instead of just continuing down after each guess. This is minor.

My biggest problem is the fact that you are making the player guess a word after each letter guess, both forcing them to lose a turn. This is not good user experience. I want the player to be able to either guess a word or a letter, not one or the other at a specific time. This needs to be changed. 

I also don't like that there is no error message when something is input incorrectly. I should see some feedback about why my input is not being accepted, rather than just repeating the prompt.

It is kind of weird to say "Leave this area blank and press enter if you do not know." and then say "That guess is not correct." because there was no guess. This needs to be changed. 

If I guess a letter correctly, and then guess the same letter again, the message given is "That guess is not correct.". The feedback should be more specific here.

I am able to input a single letter to guess the word, and even if the letter is in the word, it will be marked incorrect and I lose a turn. This is no bueno.

At the end of the game, and we did not write this story yet, but you should be more specific with the final message, ie: "You lose! The word was whatever."

File: game.rb (hunk 4 above)
Comment (NicoleCarpenter) [reply game.rb:29]: 👾
